KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift

```diff
@@ -120,17 +138,34 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
             currentMode = mode
         }
         
-        if currentMode == .earn || currentMode == .none {
-            view?.set(mode: currentMode, withSubtitle: getSubtitleText())
+        if currentMode == .burn, getBurnAmountError != nil {
+            getBurnAmountError = nil
+            updateBurnedPoints()
+        }
+        
+        view?.set(mode: currentMode, withSubtitle: getSubtitleText())
+        switch currentMode {
+        case .none:
             updateViewFromViewModel()
-        } else if currentMode == .burn, showcaseBurnModeType == .valid {
-            view?.set(mode: currentMode, withSubtitle: getSubtitleText())
-            view?.updateLoyaltyFeatures(showEarnRelatedUI: true, showBurnRelatedUI: canBurn)
-            showcaseBurnModeType = .errorInsufficientBalance
-        } else {
-            showcaseBurnMode()
+        case .earn:
+            updateViewFromViewModel()
+        case .burn:
+            self.view?.updateLoyaltyFeatures(showEarnRelatedUI: true, showBurnRelatedUI: canBurn)
         }
         
+        
+        
+//        if currentMode == .earn || currentMode == .none {
```

Comment:
Do we need to keep the commented code?

Reply:
Since it handles error showing I was meaning to replace it in the next task, the one for error handling.